microsoft/terminal
# Enable Terminal closing with ALT + F4 and warning of multiple open tabs

## Summary of the Pull Request

This PR is for: https://github.com/microsoft/terminal/issues/1589
We add a key binding called "CloseWindow", which will close the whole terminal, and warn the user if multiple tabs are opened. The user could choose to cancel or continue the closing action. 

 
## References


## PR Checklist


## Detailed Description of the Pull Request / Additional comments

In my Page-App refactoring change: https://github.com/microsoft/terminal/pull/2208, _showDialog's signature is changed. I only pass a ContentDialog to it and the contents are constructed in each specific dialog initialization methods. In this PR, I follow this pattern. 


Testing:

1. check-out this branch and build the project
2. Start terminal app
3. Open settings, add 
      {
          "command": "closeWindow",
          "keys": [
            "alt+f4"
          ]
        }
to the end of the "keybindings" list. 
3. Open multiple tabs by clicking "+"
4. Press alt + F4.
5. On the dialog, choose ""cancel", then the dialog will close and nothing happens. 
6. Press alt + F4. On the dialog, click "OK". Then the app will close. 

![image](https://user-images.githubusercontent.com/11893272/64457584-d8fbb080-d0a7-11e9-8fb6-a9f69289fa2c.png)
